org.eclipse.cdt.cross.arc.gnu/src/org/eclipse/cdt/cross/arc/gnu/ARCManagedCommandLineGenerator.java

```diff
@@ -146,31 +148,31 @@ public IManagedCommandLineInfo generateCommandLineInfo(ITool oTool, String sComm
              }
    
 
-           if (sID.indexOf(".option.target.processor") > 0) {
+           if (tcfOptionIdMatch(sID, ".option.target.processor")) {
```

Comment:
This is incorret - tcfOptionIdMatch already adds ".option.target." so you don't need to have it here. But you should leave this if/elseif as it is for know - only change usage of TCF-specific options.